src/Medidata.ZipkinTracer.Core.Collector/SpanProcessor.cs

```diff
@@ -53,71 +59,73 @@ public virtual void Stop()
+            spanProcessorTaskFactory.CreateAndStart(LogSubmittedSpans);
         }
 
         internal virtual void LogSubmittedSpans()
         {
             Span span;
-            spanQueue.TryTake(out span);
-            if (span != null)
+            while (spanQueue.TryTake(out span))
             {
-                logEntries.Add(Create(span));
+                serializableSpans.Enqueue(new SerializableSpan(span));
                 subsequentPollCount = 0;
             }
-            else if (logEntries.Any())
-            {
-                subsequentPollCount++;
-            }
+            if(serializableSpans.Count > 0) subsequentPollCount++;
 
-            if ((logEntries.Count() >= maxBatchSize)
-                || (logEntries.Any() && spanProcessorTaskFactory.IsTaskCancelled())
-                || (subsequentPollCount > MAX_NUMBER_OF_POLLS))
-            {
-                var entries = logEntries;
-                logEntries = new List<LogEntry>();
-                subsequentPollCount = 0;
-                Log(clientProvider, entries);
-            }
-        }
 
-        internal void Log(IClientProvider client, List<LogEntry> logEntries)
-        {
-            try
-            {
-                clientProvider.Log(logEntries);
-                retries = 0;
-            }
-            catch (TException)
+            if ((serializableSpans.Count() >= maxBatchSize)
+                || (serializableSpans.Any() && spanProcessorTaskFactory.IsTaskCancelled())
+                || (subsequentPollCount > MAX_NUMBER_OF_POLLS))
             {
-                if ( retries < 3 )
+                SerializableSpan serializableSpan;
+                var listOfSpans = new List<SerializableSpan>();
+                while (serializableSpans.TryDequeue(out serializableSpan))
+                {
+                    listOfSpans.Add(serializableSpan);
+                }
+                try
                 {
-                    retries++;
-                    Log(client, logEntries);
+                    if(listOfSpans.Any()) 
+                        SendSpansToZipkin(JsonConvert.SerializeObject(listOfSpans));
                 }
-                else
+                catch (WebException ex)
                 {
-                    throw;
+                    logger.Error("Failed to send to zipking with error: " + ex.Message);
                 }
             }
         }
 
-        private LogEntry Create(Span span)
+        public virtual void SendSpansToZipkin(string requestBody)
         {
-            var spanAsString = Convert.ToBase64String(ConvertSpanToBytes(span));
-            return new LogEntry()
+            using (var client = new WebClient())
             {
-                Category = "zipkin",
-                Message = spanAsString
-            };
-        }
+                var tracerPath = "/api/v1/spans";
+                UriBuilder uriBuilder = new UriBuilder();
```

Comment:
This construction of the Uri is based solely on readonly fields and string literals (that is, no change after instantiation), so it can be done at the constructor as a one-off operation (and this way you don't need to store the server and port in the class, just the constructed uri).